onnxruntime/core/graph/function_utils.cc

```diff
@@ -332,17 +333,16 @@ std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const std::string& functi
     }
   }
 
+  // model_local_functions is a member of Model instance and will be alive at the time this is invoked.
   op_schema->TypeAndShapeInferenceFunction(
-      [onnx_func_proto, func_domain_to_version, &model_local_functions](ONNX_NAMESPACE::InferenceContext& ctx) {
+      [onnx_func_proto, func_domain_to_version = std::move(func_domain_to_version), &model_local_functions](ONNX_NAMESPACE::InferenceContext& ctx) {
```

Comment:
Not sure about this `move`. We need to share the original map with all the lambdas created at this point, I think.

Reply:
May be we could pass in a shared-ptr and use it?

Reply:
I have responded offline.

Reply:
Thanks for the clarification. My fault, I was confusing the two maps (model_local_functions, and func_domain_to_version).